core/logs/src/main/scala/org/typelevel/otel4s/logs/Severity.scala

```diff
@@ -0,0 +1,170 @@
+/*
+ * Copyright 2025 Typelevel
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.typelevel.otel4s.logs
+
+import cats.Hash
+import cats.Show
+
+/** Represents the severity of a log record.
+  *
+  * @param severity
+  *   smaller numerical values correspond to less severe events (such as debug events), larger numerical values
+  *   correspond to more severe events (such as errors and critical events)
+  *
+  * @see
+  *   [[https://opentelemetry.io/docs/specs/otel/logs/data-model/#field-severitynumber]]
+  */
+sealed abstract class Severity(val severity: Int) {
+
+  override final def hashCode(): Int =
+    Hash[Severity].hash(this)
+
+  override final def equals(obj: Any): Boolean =
+    obj match {
+      case other: Severity => Hash[Severity].eqv(this, other)
+      case _               => false
+    }
+
+  override final def toString: String =
+    Show[Severity].show(this)
+
+}
+
+object Severity {
+
+  /** A fine-grained debugging event. Typically disabled in default configurations.
+    */
+  sealed abstract class Trace(severity: Int) extends Severity(severity)
+  object Trace {
+    def trace1: Trace = Trace1
+    def trace2: Trace = Trace2
+    def trace3: Trace = Trace3
+    def trace4: Trace = Trace4
```

Comment:
I feel like these methods should be on the `Severity` object for convenience, rather than nested an extra level deep

Reply:
Done